### DISCUSSION TITLE
Way to specify version of R on HPC with SSH connector?

### DISCUSSION THREAD

**Discussion by Aariq** [mschubert/clustermq#275] [Q&A] [answered]
My university's HPC currently has a bug with R 4.1 +.  I can get `clustermq` working on slurm by specifying `module load R/4.0` in the SLURM template file.  However, when I try to use the SSH connector, it's not clear to me how to specify this.  I have `module load R/4.0` in the template file, but that only affects which version of R the workers load.  Is there a way to configure the top level process with the SSH connector?

**Comment by mschubert** [comment, marked as answer]
You can provide your own [SSH template](https://github.com/mschubert/clustermq/blob/master/inst/SSH.tmpl#L5) to specify which version of R to use, or to load a module before calling R (use `&&` to chain them together):

```
ssh -o "ExitOnForwardFailure yes" -f \
    -R {{ ctl_port }}:localhost:{{ local_port }} \
    -R {{ job_port }}:localhost:{{ fwd_port }} \
    {{ ssh_host }} \
    "module load R/4.0 && R --no-save --no-restore -e \
        'clustermq:::ssh_proxy(ctl={{ ctl_port }}, job={{ job_port }})' \
        > {{ ssh_log | /dev/null }} 2>&1"
```

**Reply by Aariq** [reply]
So I would just supply the path to this template on my local machine with `options(clustermq.template = "my_ssh_template.tmpl")`?

**Reply by mschubert** [reply]
Yes, exactly

**Reply by Aariq** [reply]
I got it up and running! Knowing about that template helped me figure out how to specify a port as well.  Thanks!
